src/perl/Permabit/UserModule.pm

```diff
@@ -51,6 +72,13 @@ sub loadFromBinaryRPM {
   my $supportName = "$modFileName-support";
   my $supportFile = ($self->{useDistribution}) ? makeFullPath($self->{modDir}, "$supportName-*.rpm")
                                                : makeFullPath($topdir, "$supportName-*.rpm");
+   # if useUpstreamModule, download vdo package from distribution repo
+   if ($self->{useUpstream}) {
+     my $getFromDnf = join(' ',
+                           "dnf", "download", "--destdir", "$topdir",
+                           "$supportName");
+     $self->_step(command => $getFromDnf);
+  }
```

Comment:
Thinking about this now, I think the download of the support module probably belongs in loadFromFiles() next to the vdo package download. The main advantage to doing that is that if someone needs to change this later, it'll be easier to see that they need to change two things, since they'll be next to each other.

Reply:
This comment still applies; can you move this download step next to the other one?